### PR TITLE
Fix containerized test/build operations

### PR DESCRIPTION
**Pull request by artem-panchenko**
Currently `make test-containerized` and `make build-containerized` commands fail with the following error:

_glide install -strip-vendor -strip-vcs --cache
make: glide: Command not found
Makefile:23: recipe for target 'vendor/.up-to-date' failed
make: *** [vendor/.up-to-date] Error 127_

Fixing that by installing `glide` tool into the build container. 

Remove '-i' and '-t' flags for `docker run` in order to fix the error:

_the input device is not a TTY_

Also set correct owner to the stuff created inside container.

### CODE REVIEW

**Comment by SergK** [comment]
+1

File: Makefile [hunk 1]
```diff
@@ -48,19 +48,21 @@ release/calicoctl: clean vendor/.up-to-date
...
 build-containerized: $(BUILD_CONTAINER_MARKER)
 	mkdir -p dist
-	docker run -ti --rm --privileged --net=host \
+	docker run --rm --privileged --net=host \
 	-e PLUGIN=calico \
 	-v ${PWD}:/go/src/github.com/projectcalico/libcalico-go:rw \
 	-v ${PWD}/dist:/go/src/github.com/projectcalico/libcalico-go/dist:rw \
-	$(BUILD_CONTAINER_NAME) make bin/calicoctl
+	$(BUILD_CONTAINER_NAME) bash -c 'make bin/calicoctl; \
+        chown $(shell id -u):$(shell id -u) -R ./'
```
Comment (artem-panchenko) [inline Makefile]: done

File: Makefile [hunk 2]
```diff
@@ -48,19 +48,21 @@ release/calicoctl: clean vendor/.up-to-date
...
 # Run the tests in a container. Useful for CI, Mac dev.
 .PHONY: test-containerized
 test-containerized: $(BUILD_CONTAINER_MARKER)
-	docker run -ti --rm --privileged --net=host \
+	docker run --rm --privileged --net=host \
 	-e PLUGIN=calico \
 	-v ${PWD}:/go/src/github.com/projectcalico/libcalico-go:rw \
-	$(BUILD_CONTAINER_NAME) make ut
+	$(BUILD_CONTAINER_NAME) bash -c 'make ut; \
+        chown $(shell id -u):$(shell id -u) -R ./'
```
Comment (artem-panchenko) [inline Makefile]: done

**Comment by artem-panchenko** [comment]
Yes, it works fine for us, here is an example of `make build-containerized` cmd output: http://paste.openstack.org/show/585431/

**Comment by fasaxc** [comment]
Oops, sorry, I meant `bin/calicoctl`.  Can you try running `./bin/calicoctl` after that build?  I get a no such file or directory error from bash, which I think is due to linking against the wrong libc.

**Comment by artem-panchenko** [comment]
Shaun, I missed that, thanks! It needs /lib/ld-musl-x86_64.so.1, so on my Ubuntu 14.04 I had to install the package `sudo apt-get install musl` to execute `bin/calicoctl`. Not sure how this should be fixed, we run `calicoctl` binary inside Docker container based on `alpine-glibc` and it works just fine there.

**Comment by fasaxc** [comment]
@artem-panchenko Looks like the binary was broken before your change.  Please can you fix conflicts and then I'll merge this?  Then we can look at fixing the binary.

**Comment by robbrockbank** [comment]
@artem-panchenko :  Hi Artem - apologies for our timing on this, but we we've just done a fairly extensive history re-write which automatically closed your PR.

If you want to re-submit (sorry!) then we can merge - otherwise we'll make the same changes shortly - let me know either way.

**Comment by artem-panchenko** [comment]
@robbrockbank Hi Rob! No problem, I created another PR, please check: https://github.com/projectcalico/libcalico-go/pull/187

**Pull request closed without merging**